### PR TITLE
File based role definition documentation additions

### PR DESCRIPTION
**Pull request by jkakavas**
This commit clarifies and points out that the Role management UI and
the Role management API cannot be used to manage roles that are
defined in `roles.yml `and that file based role management is
intended to have a small administrative scope and not handle all
possible RBAC use cases.

### CODE REVIEW

**Comment by jkakavas** [comment]
c1/1 failed with the issue that https://github.com/elastic/elasticsearch/issues/46091 describes

@elasticmachine run elasticsearch-ci/1

**Comment by jkakavas** [comment]
@elasticmachine run elasticsearch-ci/1

**Review by lcawl — APPROVED** [review]
LGTM, thanks!

File: x-pack/docs/en/rest-api/security/create-roles.asciidoc [hunk 1]
```diff
@@ -24,9 +24,12 @@ privilege.
...
 
-The role API is generally the preferred way to manage roles, rather than using
-file-based role management. For more information about the native realm, see 
-{stack-ov}/realms.html[Realms] and <<configuring-native-realm>>. 
+The role management APIs are generally the preferred way to manage roles, rather than using
+{stack-ov}/defining-roles.html#roles-management-file[file-based role management]. The create
+or update roles API cannot update roles that are defined in roles files.
+
+For more information about the native realm, see
+{stack-ov}/realms.html[Realms] and <<configuring-native-realm>>.
```
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/create-roles.asciidoc]: Let's drop this sentence - index based roles aren't really related to the native realm.

File: x-pack/docs/en/rest-api/security/delete-roles.asciidoc [hunk 2]
```diff
@@ -22,10 +22,11 @@ Removes roles in the native realm.
...
 ==== {api-description-title}
 
-The Roles API is generally the preferred way to manage roles, rather than using
-file-based role management. For more information about the native realm, see 
-{stack-ov}/realms.html[Realms] and <<configuring-native-realm>>. 
+The role management APIs are generally the preferred way to manage roles, rather than using
+{stack-ov}/defining-roles.html#roles-management-file[file-based role management]. The delete roles API cannot remove roles that are defined in roles files.
 
+For more information about the native realm, see
+{stack-ov}/realms.html[Realms] and <<configuring-native-realm>>.
```
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/delete-roles.asciidoc]: As above

File: x-pack/docs/en/rest-api/security/get-roles.asciidoc [hunk 3]
```diff
@@ -23,8 +23,12 @@ privilege.
...
 ==== {api-description-title}
 
-For more information about the native realm, see 
-{stack-ov}/realms.html[Realms] and <<configuring-native-realm>>. 
+The role management APIs are generally the preferred way to manage roles, rather than using
+{stack-ov}/defining-roles.html#roles-management-file[file-based role management]. The get roles
+API cannot retrieve roles that are defined in roles files.
+
+For more information about the native realm, see
+{stack-ov}/realms.html[Realms] and <<configuring-native-realm>>.
```
Comment (tvernum) [inline x-pack/docs/en/rest-api/security/get-roles.asciidoc]: As above

**Review by tvernum — APPROVED** [review]

**Pull request merged**